tests/test_dragon_launcher.py

```diff
@@ -108,19 +116,18 @@ def is_alive(self) -> bool:
 
 def mock_dragon_env(test_dir, *args, **kwargs):
     """Create a mock dragon environment that can talk to the launcher through ZMQ"""
+    logger = logging.getLogger(__name__)
+    logging.basicConfig(level=logging.DEBUG)
+
     try:
-        context = zmq.Context()
         addr = "127.0.0.1"
         callback_port = kwargs["port"]
         head_port = find_free_port(start=callback_port + 1)
+        context = zmq.Context.instance()
```

Comment:
Not wrong just checking: I thought I heard you mention we should try and stay away from the `instance` factory method. Is this intended

Reply:
It's fine in tests (and I think I was probably wrong about the context when I discovered that the debug level dragon logs showed file permission issues).